### DISCUSSION TITLE
Error: connect EADDRNOTAVAIL 127.0.0.1

### DISCUSSION THREAD

**Discussion by VietNguyenQuoc** [getgauge/gauge#1944] [General]
This issue began after I try to delete all gauge processes by command `kill $(ps aux | grep gauge | awk '{print $2}')`

What I did:
 - Uninstall gauge cli and reinstall
 - Uninstall all the plugins and reinstall

Here's the error message:

gauge run /Users/P805877/dev/fork-SHL/app/frontend/gauge-tests/specs/personalDetails/personalDetails.spec:24 --verbose
Error: connect EADDRNOTAVAIL 127.0.0.1 - Local (0.0.0.0:53972)
at internalConnect (net.js:882:16)
at defaultTriggerAsyncIdScope (internal/async_hooks.js:294:19)
at defaultTriggerAsyncIdScope (net.js:972:9)
at process._tickCallback (internal/process/next_tick.js:61:11)
at Function.Module.runMain (internal/modules/cjs/loader.js:834:11)
at startup (internal/bootstrap/node.js:283:19)
at bootstrapNodeJSCore (internal/bootstrap/node.js:623:3)
Error ----------------------------------

[Gauge]
Error: runtime error: invalid memory address or nil pointer dereference
goroutine 1 [running]:
runtime/debug.Stack(0xc0001fb400, 0x1768da0, 0x1e95dc0)
        /opt/hostedtoolcache/go/1.13.15/x64/src/runtime/debug/stack.go:24 +0x9d
main.recoverPanic()
        /home/runner/work/gauge/gauge/gauge.go:27 +0x57
panic(0x1768da0, 0x1e95dc0)
        /opt/hostedtoolcache/go/1.13.15/x64/src/runtime/panic.go:679 +0x1b2
github.com/getgauge/gauge/runner.(*GrpcRunner).ExecuteMessageWithTimeout(0x0, 0xc0002d42c0, 0xc00025a000, 0xc000432100, 0x0)
        /home/runner/work/gauge/gauge/runner/grpcRunner.go:195 +0x22
github.com/getgauge/gauge/validation.(*SpecValidator).validateStep(0xc000432000, 0xc000238700, 0xc00047e420, 0x18)
        /home/runner/work/gauge/gauge/validation/validate.go:345 +0x1a4
github.com/getgauge/gauge/validation.(*SpecValidator).Step(0xc000432000, 0xc000238700)
        /home/runner/work/gauge/gauge/validation/validate.go:313 +0x5f1
github.com/getgauge/gauge/gauge.(*Specification).Traverse(0xc0001086c0, 0x1989fe0, 0xc000432000, 0xc0001fb8d8)
        /home/runner/work/gauge/gauge/gauge/specification.go:261 +0x256
github.com/getgauge/gauge/validation.(*SpecValidator).validate(0xc000432000, 0xc000432000, 0x13, 0x0)
        /home/runner/work/gauge/gauge/validation/validate.go:298 +0x8d
github.com/getgauge/gauge/validation.(*validator).Validate(0xc00031ba80, 0x19895c0)
        /home/runner/work/gauge/gauge/validation/validate.go:284 +0x115
github.com/getgauge/gauge/validation.ValidateSpecs(0xc00000dec0, 0x1, 0x2, 0x0, 0x0)
        /home/runner/work/gauge/gauge/validation/validate.go:174 +0x227
github.com/getgauge/gauge/execution.glob..func1(0xc00000dec0, 0x1, 0x2, 0x0)
        /home/runner/work/gauge/gauge/execution/execute.go:107 +0xef
github.com/getgauge/gauge/cmd.execute(0x1ea3260, 0xc00000dec0, 0x1, 0x2)
        /home/runner/work/gauge/gauge/cmd/run.go:244 +0x241
github.com/getgauge/gauge/cmd.glob..func13(0x1ea3260, 0xc00000dec0, 0x1, 0x2)
        /home/runner/work/gauge/gauge/cmd/run.go:93 +0x273
github.com/spf13/cobra.(*Command).execute(0x1ea3260, 0xc00000dea0, 0x2, 0x2, 0x1ea3260, 0xc00000dea0)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:854 +0x2aa
github.com/spf13/cobra.(*Command).ExecuteC(0x1ea4220, 0x102c43a, 0x1, 0x1ea3f80)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
        /home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/getgauge/gauge/cmd.Parse(0xc0001d1f18, 0xc000068750)
        /home/runner/work/gauge/gauge/cmd/cmd.go:112 +0x3d
main.main()
        /home/runner/work/gauge/gauge/gauge.go:19 +0x4c

Get Support ----------------------------
        Docs:          https://docs.gauge.org
        Bugs:          https://github.com/getgauge/gauge/issues
        Chat:          https://github.com/getgauge/gauge/discussions

Your Environment Information -----------
        darwin, 1.1.7, 5d86b72
        html-report (4.0.12), screenshot (0.0.1), ts (0.1.0)

**Comment by VietNguyenQuoc** [comment]
@sriv hello? Could you help me to resolve this issue?

**Reply by sriv** [reply]
sure, I've been trying to replicate this, but without success. From the logs it seems that the gauge-ts runner could not come up because the port is already in use? Could you see if there's a node process dangling?

**Reply by VietNguyenQuoc** [reply]
I killed all the node processes and try to run gauge again. Still failed. 
Here's the all the processes related to gauge:

ps aux | grep gauge
P805877          10831   0.0  0.0  4268448    716 s005  S+   10:45am   0:00.00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn --exclude-dir=.idea --exclude-dir=.tox gauge
P805877          10796   0.0  0.5  4939488  87100   ??  S    10:45am   0:01.76 node /Users/P805877/dev/Simple.Home.Loans.Web/app/frontend/gauge-tests/node_modules/.bin/ts-node -O {"experimentalDecorators": true,"emitDecoratorMetadata": true} -r tsconfig-paths/register -e import { GaugeRuntime }  from "gauge-ts/dist/GaugeRuntime";let runner = new GaugeRuntime();runner.start();
P805877          10789   0.0  0.5  4573332  79072   ??  S    10:45am   0:02.95 gauge daemon --lsp --dir=/Users/P805877/dev/Simple.Home.Loans.Web/app/frontend/gauge-tests

**Reply by VietNguyenQuoc** [reply]
The issue is resolved. I reinstalled the project dependencies and all good now.

**Reply by sriv** [reply]
Good to hear that. cheers!

**Reply by sriv** [reply]
also note that the gauge-ts runner process will not show up in `ps aux | grep gauge`, rather it will be a nodejs process. It's better to just get the PID of gauge process and inspect child processes. (should you need to troubleshoot this in the future)
